api/actions/serializers.py

```diff
@@ -243,12 +252,50 @@ class RegistrationActionSerializer(BaseActionSerializer):
     class Meta:
         type_ = 'registration-actions'
 
-    target = RelationshipField(
-        read_only=True,
+    permissions = ser.ChoiceField(choices=permissions.API_CONTRIBUTOR_PERMISSIONS, required=False)
+    visible = ser.BooleanField(default=True, required=False)
+    trigger = ser.ChoiceField(choices=RegistrationModerationTriggers.char_choices())
+
+    target = TargetRelationshipField(
+        target_class=Registration,
+        read_only=False,
+        required=True,
         related_view='registrations:registration-detail',
         related_view_kwargs={'node_id': '<target._id>'},
         filter_key='target__guids___id',
     )
 
-    permissions = ser.ChoiceField(choices=permissions.API_CONTRIBUTOR_PERMISSIONS, required=False)
-    visible = ser.BooleanField(default=True, required=False)
+    def create(self, validated_data):
+        trigger = validated_data.get('trigger')
+
+        target = validated_data.pop('target')
+        comment = validated_data.pop('comment', '')
+        user = validated_data.pop('user')
+
+        sanction = target.sanction
+
+        try:
+            if trigger == RegistrationModerationTriggers.SUBMIT.db_name:
+                approval_token = sanction.token_for_user(user, 'approval')
+                sanction.approve(user=user, token=approval_token)
+            elif trigger == RegistrationModerationTriggers.ACCEPT_SUBMISSION.db_name:
+                sanction.accept(user)
+            elif trigger == RegistrationModerationTriggers.REJECT_SUBMISSION.db_name:
+                sanction.reject(user=user)
+            elif trigger == RegistrationModerationTriggers.ACCEPT_WITHDRAWAL.db_name:
+                sanction.accept(user)
+            elif trigger == RegistrationModerationTriggers.REJECT_WITHDRAWAL.db_name:
+                sanction.reject(user)
+            elif trigger == RegistrationModerationTriggers.FORCE_WITHDRAW.db_name:
+                retraction = target.retract_registration(
+                    user=user, justification=comment,
+                )
+                retraction.accept(user)
+                retraction.accept(user)
```

Comment:
Have to do two approvals. One for admin contributor, one for moderator. Not in love with this solution. Working with Jon Walz on possible alternatives.